pelican/readers.py

```diff
@@ -508,17 +518,23 @@ class Readers(FileStampDataCacher):
     def __init__(self, settings=None, cache_name=""):
         self.settings = settings or {}
         self.readers = {}
+        self.disabled_readers = {}
+        # extension => reader for readers that are enabled
         self.reader_classes = {}
+        # extension => reader for readers that are not enabled
+        disabled_reader_classes = {}
 
         for cls in [BaseReader] + BaseReader.__subclasses__():
             if not cls.enabled:
                 logger.debug(
                     "Missing dependencies for %s", ", ".join(cls.file_extensions)
                 )
```

Comment:
It looks like this piece of code is already doing some of the work for you; why not hang the warning off of here?

Reply:
It's easy to warn if disabled, but if I have no relevant files, I don't want the warning. 

So I need somewhere that knows there are relevant files, which mostly in the other place. 

If you think this can look at file suffixes, please say more.

Reply:
Ah... I see what you're getting at. 

I think I'd prefer a single warning rather than one per file, but I'm not dead set on it either way, and I'd rather have lots of warnings than none, so ... +1 from me.

Reply:
I agree one would be better. I could track whether a particular suffix has been warned about and suppress the rest, but I'd rather have fewer lines of code devoted to this minor feature.